packages/cli-repl/src/mongosh-repl.ts

```diff
@@ -163,12 +163,12 @@ class MongoshNodeRepl implements EvaluationListener {
   ioProvider: MongoshIOProvider;
   onClearCommand?: EvaluationListener['onClearCommand'];
   insideAutoCompleteOrGetPrompt: boolean;
-  inspectCompact: number | boolean = 0;
-  inspectDepth = 0;
+  inspectCompact?: number | boolean = 0;
+  inspectDepth?: number = 0;
   started = false;
-  showStackTraces = false;
+  showStackTraces?: boolean = false;
   loadNestingLevel = 0;
-  redactHistory: 'keep' | 'remove' | 'remove-redact' = 'remove';
+  redactHistory?: 'keep' | 'remove' | 'remove-redact' = 'remove';
```

Comment:
Let’s maybe take care of this in the place where these properties are set, rather than here, giving the wrong impression that they can ever actually become `undefined`?

i.e. either

```js
    this.inspectCompact = (await this.getConfig('inspectCompact'))!;
```

(although the linter warns)

or 

```js
    const inspectCompact = await this.getConfig('inspectCompact')
    if (inspectCompact)
      this.inspectCompact = inspectCompact;
```

(verbose but a maybe a bit too defensive)